app/fragments/apps/AppFragment.tsx

```diff
@@ -20,9 +23,21 @@ export const AppFragment = fragment(() => {
     const dark = c.isDark();
     const fontColor = dark ? '#fff' : '#000';
     const key = engine.products.keys.getDomainKey(domain);
+
+    // Should never happen
     if (!appData) {
-        throw Error('No App Data');
+        warn('No App Data');
+        navigation.goBack();
+        return null;
+    }
+
+    // Should never happen
```

Comment:
So why is can be nullable?

Reply:
It can be null returned from the persistence storage, I thought it would be bad if we throw an error that crashes the app on navigation. So checking for null is a bit nicer from the ui stand point, dont you think? I've simplified return statement.